backend/admin_digest_router.py

```
@@ -0,0 +1,141 @@
+    return hmac.compare_digest(provided.encode("utf-8"), expected.encode("utf-8"))
+
+
+def _build_payload_or_500():
+    """Returns a DigestPayload (or raises HTTPException(503/500))."""
+    try:
+        from services.daily_digest import build_daily_digest
+        from services.decisions_store import get_decisions_store
+        from services.workflow_orchestrator import get_workflow_orchestrator
+    except ImportError as exc:
+        log.exception("admin_digest.import_failed exc=%s", exc)
+        raise HTTPException(status_code=503, detail=f"digest dependencies not importable: {exc}")
```

Ruff (0.15.17)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)